espnet2/asr/encoder/e_branchformer_encoder.py

```diff
@@ -386,6 +397,8 @@ def forward(
         xs_pad: torch.Tensor,
         ilens: torch.Tensor,
         prev_states: torch.Tensor = None,
+        ctc: CTC = None,
+        layer: int = None,
```

Comment:
I think the name `layer` is a bit hard to understand. We can call it `max_layers`?